File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/PreviewVideoDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * DTO for a preview video from Udemy course preview page.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PreviewVideoDTO {
    private Long id;
    private String title;
    private String thumbnailUrl;
    private String contentSummary; // e.g., "02:22"
    private Integer timeEstimation; // in seconds
    private String videoUrl; // HLS m3u8 URL from media_sources
    private List<VideoSourceDTO> streamUrls; // MP4 files at different resolutions
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/PreviewVideoDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for a preview video from Udemy course preview page.
 */
data class PreviewVideoDTO(
    val id: Long? = null,
    val title: String? = null,
    val thumbnailUrl: String? = null,
    // e.g., "02:22"
    val contentSummary: String? = null,
    // in seconds
    val timeEstimation: Int? = null,
    // HLS m3u8 URL from media_sources
    val videoUrl: String? = null,
    // MP4 files at different resolutions
    val streamUrls: List<VideoSourceDTO>? = null,
)
